flow/project.py

```diff
@@ -2717,11 +3041,54 @@ def _verify_group_compatibility(self, groups):
         return all(a.isdisjoint(b) for a in groups for b in groups if a != b)
 
     def _verify_aggregate_project(self, aggregate):
-        """Verifies that all aggregates belongs to the same project."""
-        for job in aggregate:
-            if job not in self:
-                raise ValueError("Job {} is not present "
-                                 "in the project".format(job))
+        """Verifies that the aggregate belongs to the this project."""
+        return any(aggregate in stored_aggregator for stored_aggregator in self._stored_aggregators)
+
+    def _verify_aggregate_eligibility(self, aggregate, jobs):
+        """Verifies whether the aggregate is present in the jobs provided by the users
+        or not.
+        """
+        if jobs is None:
+            return True
+        return aggregate in jobs
+
+    def _get_aggregate_from_id(self, id):
+        # Iterate over all the instances of stored aggregates and search for the
+        # aggregate in those instances.
+        for stored_aggregator in self._stored_aggregators:
+            try:
+                return stored_aggregator[id]
+            except LookupError:  # Didn't found aggregate in this stored object
+                pass
+        # Raise error as didn't found the id in any of the stored object
+        raise LookupError(f"Did not find aggregate having id {id} in the project")
+
+    def _convert_aggregates_from_jobs(self, jobs):
+        # The jobs parameter in the public methods like ``run``, ``submit``, ``status``
+        # may accept either a signac job or an aggregate.
+        # We convert that job / aggregate (which may be of any type (eg. list)) to an
+        # aggregate of type ``tuple``
+
+        if jobs is not None:
+            aggregates = set()  # Set in order to prevent duplicate entries
+            for aggregate in jobs:
+                # User can still pass signac jobs
+                if isinstance(aggregate, signac.contrib.job.Job):
+                    if aggregate not in self:
+                        raise LookupError(f"Did not find the job {aggregate} in the project")
+                    aggregates.add((aggregate,))
+                else:
+                    try:
+                        aggregate = tuple(aggregate)
+                        if not self._verify_aggregate_project(aggregate):
+                            raise LookupError(f"Did not find aggregate {aggregate} in the project")
+                        aggregates.add(aggregate)  # An aggregate provided by the user
+                    except TypeError:
```

Comment:
It seems this try block is only meant to catch errors in the `tuple` call. If that is the case we should use the `try`, `except`, `else` construct. Or just write the code that would be in the `else` block outside the try block.